Makefile

```diff
@@ -70,5 +70,9 @@ rye-build: rye-src
 	export GO111MODULE=${GO111MODULE} && \
 		cd rye-src && go build -x -tags "b_tiny" -o rye .
 
+.PHONY: haskell
+haskell:
+	echo "AAABBAAC" | runhaskell src/haskell/Compress_turbomack.hs
```

Comment:
using interpreted mode (ghci) since there seems to be no reason to try to produce machine code for something like this. Just keep in mind in case you consider bench-marking this - it should be turned to binary first.

Reply:
I'm not entirely sure if we actually should do benchmarking across all solutions. Especially due to very different use-cases that these languages might have. There is an attempt to compare [Python solutions here](https://github.com/otobrglez/compression-puzzle/pull/14) and [F# here](https://github.com/otobrglez/compression-puzzle/blob/master/src/fsharp/benchmark.fsx) so perhaps comparing different Haskell solutions among themselves? :)